### PR TITLE
CB-8970 entitlement based upscale and downscale on specific services.…

### PR DESCRIPTION
**Pull request by doktoric**
… We need to request new ums entitlements as well for this feature to use. Mostly for nifi team to use but I prepared for streaming and other scenarios as well.

See detailed description in the commit message.

### CODE REVIEW

File: mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/grpc/service/auth/MockUserManagementService.java [hunk 1]
```diff
@@ -594,6 +598,8 @@ public void getAccount(GetAccountRequest request, StreamObserver<GetAccountRespo
                                 .addEntitlements(createEntitlement(DATAHUB_AZURE_AUTOSCALING))
                                 .addEntitlements(createEntitlement(DATAHUB_AWS_AUTOSCALING))
                                 .addEntitlements(createEntitlement(LOCAL_DEV))
+                                .addEntitlements(createEntitlement(DATAHUB_FLOW_SCALING))
+                                .addEntitlements(createEntitlement(DATAHUB_STREAMING_SCALING))
```
Comment (biharitomi) [inline mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/grpc/service/auth/MockUserManagementService.java:602]: Shouldn't the `DATAHUB_DEFAULT_SCALING` should also be added here?

File: core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterCommonService.java [hunk 2]
```diff
@@ -127,7 +129,14 @@ private FlowIdentifier clusterHostgroupAdjustmentChange(Long stackId, UpdateClus
                     updateJson.getHostGroupAdjustment().getHostGroup(), stack.getName()));
         }
         if (blueprintService.isClouderaManagerTemplate(blueprint)) {
-            cmTemplateValidator.validateHostGroupScalingRequest(blueprint, hostGroup.get(),
+            User creator = stack.getCreator();
+            String userCrn = creator.getUserCrn();
```
Comment (biharitomi) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterCommonService.java:133]: Shouldn't we gather and check the user who initiated this scaling operation?

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java [hunk 3]
```diff
@@ -80,7 +81,12 @@ public void validateServiceRoles(Stack stack, InstanceGroupAdjustmentV4Request i
                 .filter(e -> e.getName().equals(instanceGroup))
                 .findFirst();
         if (hostGroup.isPresent()) {
+            User creator = stack.getCreator();
+            String userCrn = creator.getUserCrn();
```
Comment (biharitomi) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java:85]: Shouldn't we gather and check the user who initiated this scaling operation?

**Review by biharitomi — CHANGES_REQUESTED** [review]

File: mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/grpc/service/auth/MockUserManagementService.java (hunk 1 above)
Comment (doktoric) [reply mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/grpc/service/auth/MockUserManagementService.java:602]: I dont want to enable that one yet

File: core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterCommonService.java (hunk 2 above)
Comment (doktoric) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterCommonService.java:133]: I think what we want to check is the account entitlement. It does not matter who initiated this

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java (hunk 3 above)
Comment (doktoric) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java:85]: nope

File: mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/grpc/service/auth/MockUserManagementService.java (hunk 1 above)
Comment (biharitomi) [reply mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/grpc/service/auth/MockUserManagementService.java:602]: ok

File: core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterCommonService.java (hunk 2 above)
Comment (biharitomi) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterCommonService.java:133]: Yet, but probably it we should consider it later, I am ok with that, but I said :)

**Review by biharitomi — APPROVED** [review]

**Pull request merged**